File: 251118/요일 맞추기/guess-day-of-week.py
m1, d1, m2, d2 = map(int, input().split())

month = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
week = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']

def days(m, d):
    day = 0
    for i in range(1, m):
        day += month[m]
    day += d
    return day

result = days(m2, d2) - days(m1, d1)
print(week[(result + 7) % 7])

File: 260304/요일 맞추기/guess-day-of-week.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");
const [m1, d1, m2, d2] = input[0].split(" ").map(Number);

const days = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
const date = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"];

let diff = 0;

if (m1 === m2) {
    diff = d2 - d1;
} else {
    // 시작 달
    diff += (days[m1] - d1);

    for (let m = m1 + 1; m <= m2 - 1; m++) {
        diff += days[m];
    }

    // 끝나는 달
    diff += d2;
}

const answer = date[((diff % 7) + 7) % 7];
console.log(answer);
